src/video-player.js

```diff
@@ -274,11 +275,36 @@ class VideoPlayer extends Utils.mixin(Eventable) {
     this.videojs.ready(() => {
       onReady();
 
+
       if (ready) {
         ready(this);
       }
     });
 
+
+    this.nbCalls = 0;
+    this.reTryVideo = (maxNumberOfCalls, timeout) => {
+      if (!this.isVideoReady()) {
+        if (this.nbCalls < maxNumberOfCalls) {
+          this.nbCalls++;
+          this.videojs.setTimeout(this.reTryVideo, timeout);
+        } else {
+          let e = new Error('Video is not ready please try later');
+          this.videojs.trigger('error', e);
+        }
+      }
+
+    };
+
+    this.isVideoReady = () => {
+      let s = this.videojs.readyState();
+       if (s === 4) {
```

Comment:
Also, there's something wrong in the indentation here (`let` and `if` are not aligned).